crates/build/src/lib.rs

```diff
@@ -56,13 +57,26 @@ where
         load_and_validate(root_url.clone(), root_spec, fetcher, drivers, &config).await;
     all_tables.meta.insert_row(config.clone());
 
+    let has_typescript_derivations = all_tables
+        .derivations
+        .iter()
+        .any(|derivation| derivation.typescript_module.is_some());
+
+    let has_npm_resources = all_tables
+        .resources
+        .iter()
+        .any(|resource| resource.content_type == protocol::flow::ContentType::TypescriptModule);
+
+    let typescript_enabled = has_typescript_derivations || has_npm_resources;
+
     // Output database path is implied from the configured directory and ID.
     let output_path = directory.join(&config.build_id);
     let db = rusqlite::Connection::open(&output_path).context("failed to open catalog database")?;
 
     // Generate TypeScript package? Generation should always succeed if the input catalog is valid.
     if all_tables.errors.is_empty()
         && (config.typescript_generate || config.typescript_compile || config.typescript_package)
+        && typescript_enabled
```

Comment:
nit on naming: I read this condition first, and I thought there is a flag somewhere for enabling typescript, and then I read the definition and realised this means that there are either some derivations or npm resources that need to be built, maybe we can rename this to give that message, something like `has_typescript_resources`?

Reply:
Yup, great point. Will do